assembly/src/parsers/serde/serialization.rs

```diff
@@ -425,8 +425,43 @@ impl Serializable for Instruction {
             Self::CDropW => target.write_opcode(OpCode::CDropW),
 
             // ----- input / output operations --------------------------------------------------------
-            Self::PushConstants(values) => {
-                target.write_opcode(OpCode::PushConstants);
+            Self::PushU8(value) => {
+                target.write_opcode(OpCode::PushU8);
+                target.write_u8(*value);
+            }
+            Self::PushU16(value) => {
+                target.write_opcode(OpCode::PushU16);
+                target.write_u16(*value);
+            }
+            Self::PushU32(value) => {
+                target.write_opcode(OpCode::PushU32);
+                target.write_u32(*value);
+            }
+            Self::PushFelt(value) => {
+                target.write_opcode(OpCode::PushFelt);
+                target.write_felt(*value);
+            }
+            Self::PushWord(values) => {
+                target.write_opcode(OpCode::PushWord);
+                values.iter().for_each(|&v| target.write_felt(v));
```

Comment:
nit: I wonder why clippy didn't mention we are passing a ref of a ref. By default, `iter` will always yield refs, so if we `&v` here, we are in fact passing `&&Felt` to `write_felt`. Its ok because it will deref to `&Felt` anyway, but this might give the impression that we are trying to do something else

Reply:
I'm not sure how it works, but it looks like when we passing `&v` to `for_each`, it evaluates whole `&v` expression as `&Felt`, that's why I can provide into `write_felt` required plain `v`, because it's just `Felt`. When I pass to `for_each` just `v`, it's evaluated as `&Felt`, so I have to provide `*v` to the `write_felt` to dereference it

Reply:
`iter` usually will iterate with references as items, and `into_iter` will create owned items iterators. Since we are using the former, I'm assuming it is with `&Felt`. If its owned Felt, then ignore this comment